sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -577,7 +577,15 @@ export class EventHubSender extends LinkEntity {
           this._sender!.on(SenderEvents.rejected, onRejected);
           this._sender!.on(SenderEvents.modified, onModified);
           this._sender!.on(SenderEvents.released, onReleased);
-          waitTimer = setTimeout(actionAfterTimeout, Constants.defaultOperationTimeoutInSeconds * 1000);
+          waitTimer = setTimeout(
+            actionAfterTimeout,
+            options == undefined ||
+              options.retryOptions == undefined ||
+              options.retryOptions.operationTimeoutInMs == undefined ||
+              options.retryOptions.operationTimeoutInMs <= 0
+              ? Constants.defaultOperationTimeoutInSeconds * 1000
+              : options.retryOptions.operationTimeoutInMs
```

Comment:
- For readability, pull this code out and have a variable/constant hold the value of the timeout
- Initializing `options` with `{}` is done early on, so the check `options == undefined` is not needed

Reply:
Within the event handler it seemed to require the check, please see updated code